### DISCUSSION TITLE
Alembic.command.check (v.1.9.0)

### DISCUSSION THREAD

**Discussion by HusseinWarsame** [sqlalchemy/alembic#1179] [Usage Questions] [answered]
Hello,

My team has a few redundant operations in the revision history of some our projects. I'd like to remove them since it is conflicting with the adoption of a naming convention but have encountered some unexpected errors.

For example,

![image](https://user-images.githubusercontent.com/21069829/218887022-ac62dd5f-c611-4674-a1ba-6d0ba23dd1a2.png)

Here we have the addition and removal of the same foreign key on two different tables. My expectation was that the "sum" of these operations would not count as a pending upgrade operation. This is because when I try to generate a migration with Flask-Migrate for these pending upgrade operations, there are no changes in the schema detected:

![image](https://user-images.githubusercontent.com/21069829/218887070-b1c89c69-d344-4d4e-a0c6-8c6a4c3bc0f4.png)

Is this expected behaviour for the `check` command?

**Comment by zzzeek** [comment, marked as answer]
the check command should produce the same result as a plain autogenerate command, they run the same comparison routine.   there is also no consideration of actual upgrade operations during autogenerate or check, it strictly looks at the MetaData collection passed and compares its contents to what reflection sees.

**Reply by HusseinWarsame** [reply]
I'm not sure I understand. Flask-Migrate is using `alembic revision --autogenerate` to detect changes to the schema. From the [Flask-Migrate docs on using alembic check](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#running-alembic-check-to-test-for-new-upgrade-operations): 

> ...it returns an error code plus a message if it is detected that new operations would be rendered into a new revision, or if not, returns a success code plus a message. When alembic check returns a success code, this is an indication that the alembic revision --autogenerate command would produce only empty migrations, and does not need to be run.

In my case, we have an error code when operations are not being rendered into a new revision:

![image](https://user-images.githubusercontent.com/21069829/219100858-7e9a6522-2391-4ecf-90be-3331a4b2f049.png)

Am I misunderstanding the Flask-Migrate docs on what `check` does?

**Reply by zzzeek** [reply]
if you run the "check" operation programmatically without using the alembic command line, it will raise that error.  the command line catches all exceptions and converts them to nonzero exit codes, unless the ``--raiseerr`` parameter is passed.   obviously I can't speak for what flask-migrate is doing, perhaps there's an issue you should report to that project.
